src/plugins/typechecker/typechecker/simplespecification.ini

```diff
@@ -0,0 +1,26 @@
+#@META elektra/spec/type = RegexContains b a => Key b :: . -> Key a -> Key a
+#@META elektra/spec/impl = fallback a (Key P.Nothing) = a \n fallback _ a = a
+[/elektra/spec/fallback/#]
+
+#@META elektra/spec/type = RegexContains b a => Key b :: . -> Key a -> Key a
+#@META elektra/spec/impl = override (Key P.Nothing) b = b \n override a _ = a
+[/elektra/spec/override/#]
+
+#@META elektra/spec/type = RegexIntersects a b => P.Proxy b :: Range . -> Key a -> Key (RegexIntersection a b)
+#@META elektra/spec/impl = checkrange _ a
+[/elektra/spec/check/range]
+
+#@META elektra/spec/type = RegexIntersects a "-[1-9]|-214748364[0-8]|-?[1-9][0-9]|-?[1-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?1[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?20[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?21[0-3][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?214[0-6][0-9][0-9][0-9][0-9][0-9][0-9]|-?2147[0-3][0-9][0-9][0-9][0-9][0-9]|-?21474[0-7][0-9][0-9][0-9][0-9]|-?214748[0-2][0-9][0-9][0-9]|-?2147483[0-5][0-9][0-9]|-?21474836[0-3][0-9]|[0-9]|214748364[0-7]" => Key a -> Key (RegexIntersection a "-[1-9]|-214748364[0-8]|-?[1-9][0-9]|-?[1-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?1[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?20[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?21[0-3][0-9][0-9][0-9][0-9][0-9][0-9][0-9]|-?214[0-6][0-9][0-9][0-9][0-9][0-9][0-9]|-?2147[0-3][0-9][0-9][0-9][0-9][0-9]|-?21474[0-7][0-9][0-9][0-9][0-9]|-?214748[0-2][0-9][0-9][0-9]|-?2147483[0-5][0-9][0-9]|-?21474836[0-3][0-9]|[0-9]|214748364[0-7]")"
+#@META elektra/spec/impl = checklong a = a
+[/elektra/spec/check/long]
+
+#@META check/range = 0-5000
+[key1]
+
+#@META check/range = 7200-10000
+#@META override/#1 = key1
+[key2]
+
+#@META check/long =
+#@META fallback/#1 = key1
+[key3]
```

Comment:
Why is there no validation example? Examples with regex would be the most obvious ones.

An example with enums would be interested, too. (Or isn't this implemented now?)

Reply:
not now. this will be handled with the other pr, the regex dispatcher that preprocesses enum keywords and things like that. i may add one over there though